man/coap_pdu_setup.txt.in

```diff
@@ -331,30 +377,101 @@ some cases the maximum data size of an option can be exceeded if not stripped
 and hence be illegal.  This is done by using coap_encode_var_safe() or
 coap_encode_var_safe8().
 
+*Function: coap_encode_var_safe()*
+
 The *coap_encode_var_safe*() function encodes _value_ into _buffer_ which has
 a size of _size_ in bytes.  Normally, the _buffer_ size should be at least
 the sizeof(int) bytes unless you definitely know less space is required.
 
+*Function: coap_encode_var_safe8()*
+
 The *coap_encode_var_safe8*() function encodes 8 byte _value_ into _buffer_
 which has a size of _size_ in bytes.  Normally, the _buffer_ size should be at
 least 8 bytes unless you definitely know less space is required.
 
+*Function: coap_split_path()*
+
 The *coap_split_path*() function splits up _path_ of length _length_ and
-places the result in _buffer_ which has a size of _buflen_.  _buflen_ needs
+places the result in _buffer_ which has a size of _buflen_ with the NULL
+character separating each path component.  _buflen_ needs
 to be preset with the size of _buffer_ before the function call, and then
-_buflen_ is updated with the actual size of _buffer_ used.
+_buflen_ is updated with the actual size of _buffer_ used. The return
+value indicates the number of components that individual COAP_OPTION_URI_PATH
+options need to be created for.
+
+*Function: coap_split_query()*
 
 The *coap_split_query*() function splits up _query_ of length _length_ and
-places the result in _buffer_ which has a size of _buflen_.  _buflen_ needs
+places the result in _buffer_ which has a size of _buflen_ with the NULL
+character separating each path component.  _buflen_ needs
 to be preset with the size of _buffer_ before the function call, and then
-_buflen_ is updated with the actual size of _buffer_ used.
+_buflen_ is updated with the actual size of _buffer_ used. The return
+value indicates the number of components that individual COAP_OPTION_URI_QUERY
+options need to be created for.
+
+PDU OPTIONS - LIBCOAP HANDLING
+------------------------------
+Most of the options are under the control of the applications, but the
+following are primarily used internally by libcoap.
+
+*COAP_OPTION_BLOCK1* and *COAP_OPTION_BLOCK2*
+
+These BLOCK options are used when a large body needs to be split up into
+multiple payloads.  Following the introduction of
+*coap_context_set_block_mode*(3), libcoap can internally handle the setting
+of these options (see *coap_block*(3)). Applications can continue to include
+these options to set hint block size values.
+
+It is recommended that
+*coap_context_set_block_mode(context, COAP_BLOCK_USE_LIBCOAP|COAP_BLOCK_SINGLE_BODY)*
+is used to reduce the programming requirements for block handling within
+the applications.
+
+*COAP_OPTION_ECHO*
+
+This option can be set by the server application to indicate that the state of
+the client's freshness is confirmed. The libcoap client logic will detect the
+use of the ECHO option by the server and reflect back the ECHO value in the
```

Comment:
I did not check the Echo option but usually, the CoAP option names only have every word's first letter capitalized, the other letters are lowercase.

Reply:
Corrected this and other option names in the document.  Updated code pushed.

Reply:
Thanks!